src/test/java/com/rhc/sonarqube/auth/openshift/test/MockHttpClient.java

```diff
@@ -46,8 +46,8 @@ public void close() throws IOException {
 	@Override
 	public Response execute(String userAgent, Map<String, String> headers, Verb httpVerb, String completeUrl,
 			byte[] bodyContents) throws InterruptedException, ExecutionException, IOException {
-		
-		if(completeUrl.contentEquals("http://127.0.0.1/oapi/v1/users/~")) {
+
+		if(completeUrl.contentEquals("http://127.0.0.1/apis/user.openshift.io/v1/users/~")) {
```

Comment:
This API endpoint is very OpenShift specific. Would it be possible to make it more generic to Kubernetes, or would that be a functionality beyond what native K8s provides?

Reply:
There really isn't. The first url I was using was still OpenShift specific but the whole implementation was not correct. Kube has an authorization API but it requires giving access to cluster resources. So, if I wanted it to work with that API I would have to make an api call that would be like:

Can user1 view pods (great you're a sonar user)
Can user1 create pods (great you're a sonar admin)

I am thinking that approach requires too much cluster access. Also, differentiating user and admin access would be awkward. Instead using the group approach - a group can be created specifically for the application and sonar queries user information for groups that the user belongs to. Here - the user can essentially have access to nothing in OpenShift and still use and administer Sonarqube